src/lib/Lib/CMakeExecution.cpp

```diff
@@ -0,0 +1,164 @@
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (c) 2024 Fernando Pelliccioni (fpelliccioni@gmail.com)
+//
+// Official repository: https://github.com/cppalliance/mrdocs
+//
+
+#include "lib/Lib/CMakeExecution.hpp"
+
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/MemoryBuffer.h>
+#include <llvm/Support/Path.h>
+#include <llvm/Support/Program.h>
+
+namespace clang {
+namespace mrdocs {
+
+namespace {
+
+Expected<std::string>
+getCmakePath() {
+    auto const path = llvm::sys::findProgramByName("cmake");
+    MRDOCS_CHECK(path, "CMake executable not found");
+    std::optional<llvm::StringRef> const redirects[] = {llvm::StringRef(), llvm::StringRef(), llvm::StringRef()};
+    std::vector<llvm::StringRef> const args = {*path, "--version"};
+    int const result = llvm::sys::ExecuteAndWait(*path, args, std::nullopt, redirects);
+    MRDOCS_CHECK(result == 0, "CMake execution failed");
+    return *path;
+}
+
+Expected<std::string>
+executeCmakeHelp(llvm::StringRef cmakePath)
+{
+    llvm::SmallString<128> outputPath;
+    MRDOCS_CHECK(!llvm::sys::fs::createTemporaryFile("cmake-help", "txt", outputPath), 
+        "Failed to create temporary file");
+    std::optional<llvm::StringRef> const redirects[] = {llvm::StringRef(), outputPath.str(), llvm::StringRef()};
+    std::vector<llvm::StringRef> const args = {cmakePath, "--help"};
+    llvm::ArrayRef<llvm::StringRef> emptyEnv;
+    int const result = llvm::sys::ExecuteAndWait(cmakePath, args, emptyEnv, redirects);
+    MRDOCS_CHECK(result == 0, "CMake execution failed");
+
+    auto const bufferOrError = llvm::MemoryBuffer::getFile(outputPath);
```

Comment:
I don't know the type of `bufferOrError`, but if it's a common type this is a job for MRDOCS_TRY.

Reply:
I can't use `MRDOCS_TRY` because apparently it uses `std::move` and the type returned by `llvm::sys::ExecuteAndWait()` is not movable.

Reply:
I will mark this comment as resolved.